yarl/_url.py

```diff
@@ -713,6 +713,7 @@ def _validate_authority_uri_abs_path(host, path):
 
     def _make_child(self, segments, encoded=False):
         """add segments to self._val.path, accounting for absolute vs relative paths"""
+        need_trailing_slash = segments and segments[-1].endswith("/")
         parsed = []
```

Comment:
Instead of a separate flag, just set `parsed` to `[""]`:

```suggestion
        # keep the trailing slash if the last segment ends with /
        parsed = [""] if segments and segments[-1][-1:] == "/" else []
```

Reply:
OK, done.